documents/compute_fields.go

```diff
@@ -20,39 +23,45 @@ const (
 
 	// ErrComputeFieldsComputeNotFound is a sentinel error when WASM doesn't expose 'compute' function
 	ErrComputeFieldsComputeNotFound = errors.Error("'compute' function not exported")
+
+	// computeFieldsTimeout is the max time we let the WASM computation to be run.
```

Comment:
why 30 secs? out of curiousity

Reply:
should be good enough time to run a Complex WASM function as well

Reply:
Do you know our p2p timeout value? as long as this value is smaller than that one we are good. Do you think we should have it configurable? or no need for now.

Reply:
This should not be configurable IMO. since this is core consensus, if configured differently on different nodes, might lead to timeouts and consensus being failed